app/screens/channel/channel_post_list/intro/public_or_private_channel/index.ts

```diff
@@ -31,11 +31,15 @@ const enhanced = withObservables([], ({channel, database}: {channel: ChannelMode
         const me = database.get<SystemModel>(SYSTEM).findAndObserve(SYSTEM_IDENTIFIERS.CURRENT_USER_ID).pipe(
             switchMap(({value}) => database.get<UserModel>(USER).findAndObserve(value)),
         );
+        const profile = channel.observe().pipe(
+            switchMap(() => channel.creator.observe()),
+            catchError(() => of$(undefined)),
+        );
```

Comment:
Im not really a fan of this and the fix seems incomplete, I would like to see that this observable eventually ends up observing the user and I expect the component to fetch the user if it is not found in the database.

Where I differ from @larkox here is that I rather have the creator fetched on a need basis than fetching them all as they could be too many and without much need to get them.

Reply:
We are already fetching the channel's creator if its value is empty here : https://github.com/mattermost/mattermost-mobile/blob/f309fa98ee9687806a3d29e78edbd678c1e5f502/app/screens/channel/channel_post_list/intro/public_or_private_channel/public_or_private_channel.tsx#L70

Reply:
Yes Avinash we are fetching the channel creator there, but this is not updating.

Channel already has set `creatorId` and that field nor anything in the channel record is being updated, so the observer for `channel` does not trigger thus `profile` will be `undefined` at all times until the next mount!

in order to properly fix this you should use something like this:
```ts
const profile = channel.observe.pipe(
    switchMap((c) => observeUser(database, c.creatorId),
);
```

Why?

well `observeUser` actually returns an observer on the user or undefined but it also keeps observing on the query for that userId, so as soon as the new record is created, the query returns the new value.

get it ?

Reply:
Thanks a lot @enahum .  I totally missed this part.... observing a channel is not the same as observing the user record getting updated.

I have made some modification to your suggestion and also tested it against the 'staff' team on the community server.  Now, when we don't have the channel creator, it fetches it ( as usual ) but now, it also updates the component with this new value.

![channel creator](https://user-images.githubusercontent.com/43164508/160557335-09092f09-4297-4b97-889d-ee629d30e6ca.png)